cmd/git-sync/main.go

```diff
@@ -460,11 +430,13 @@ func main() {
 			if *flMaxSyncFailures != -1 && failCount >= *flMaxSyncFailures {
 				// Exit after too many retries, maybe the error is not recoverable.
 				log.Error(err, "too many failures, aborting", "failCount", failCount)
+				exportError(err.Error())
```

Comment:
In these cases - is the Error() enough context?  This could come as "File does not exist", which is not super helpful.

I think this is worth a bit more scaffolding.  What do you think of something like:

```
type customLogger struct {
    logr.Logger
    errorFile string
}

func (l customLogger) Error(err error, msg string, kvList ...interface{}) {
    l.Logger.Error(err, msg, kvList...)
    if l.errorFile != "" {
        l.writeErrorFile(err, msg, kvList...)
    }
}
```

and instead of `var log = glogr.New()` at global scope:

```
var log customLogger
```

and in main():

```
log = customLogger{ glogr.New(), *flErrorFile }
```

Then you can totally encapsulate *flErrorFile in `main()` and `writeErrorFile()` and `deleteErrorFile()` can be methods on `customLogger`.

Within `writeErrorFile()` you can unpack the kvList into something like:

```
    payload := struct {
        Msg string
        Err string
        Args map[string]interface{}
    } {
        Msg: msg,
        Err: err.Error(),
        Args: map[string]interface{}{}
    }
    for i := 0; i < len(kvList); i++ {
        // unpack into Args
    }
  }
```

Then when when you write to the file:

```
jb, err := json.Marshall(payload)
```

So we write something with a defined structure to the file.

Reply:
Thanks for the suggestion. I like the defined structure in the error file where we can have additional information.
DONE